src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.VectorSearch.cs

```diff
@@ -0,0 +1,445 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Corax.Utils;
+using Jint;
+using Jint.Native;
+using Raven.Client.Documents.Indexes.Vector;
+using Raven.Server.Documents.Indexes.Persistence.Corax;
+using Raven.Server.Documents.Indexes.Static.JavaScript;
+using Raven.Server.Documents.Indexes.VectorSearch;
+using Sparrow;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Indexes.Static;
+
+public partial class AbstractStaticIndexBase
+{
+    /// <summary>
+    /// Dictionary training process occurs in IsOnBeforeExecuteIndexing. Since we're not training dictionaries with vectors, and considering computation
+    /// power required (e.g., generating embeddings from text), it is better to skip that part as there is no benefit in performing it.
+    /// </summary>
+    /// <param name="currentIndexingScope">Current indexing scope.</param>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static bool IsDictionaryTrainingPhase(CurrentIndexingScope currentIndexingScope)
+    {
+        return currentIndexingScope != null && currentIndexingScope.Index.IsOnBeforeExecuteIndexing;
+    }
+
+    internal static IndexField RetrieveVectorField(string fieldName, object value)
```

Comment:
Extracted method from `CreateVector`. It was extracted to have common logic in Jint/C# converter.